src/System.Drawing.Common/src/System/Drawing/Brush.cs

```diff
@@ -100,34 +60,13 @@ protected virtual void Dispose(bool disposing)
                     Debug.Assert(status == SafeNativeMethods.Gdip.Ok, "GDI+ returned an error status: " + status.ToString(CultureInfo.InvariantCulture));
 #endif
                 }
-                catch (Exception ex)
-                {
-                    if (ClientUtils.IsSecurityOrCriticalException(ex))
-                    {
-                        throw;
-                    }
-
-                    Debug.Fail("Exception thrown during Dispose: " + ex.ToString());
-                }
```

Comment:
What if it gets another runtime-related exception type, like EntryPointNotFound for the P/Invoke?  Or what about when Mono wants to use this implementation rather than the current one?